feat: Add tags to Units

yusuf-musleh opened the pull request:
### Description
Implements tagging Units in the sample courses. This should help with testing features/functionality related to tags on units.

### Related Tickets:
- https://github.com/openedx/modular-learning/issues/118

### Testing Instructions
Similar to what is mentioned in the README:
1. To begin, clone this repo inside a directory so it can be accessed from within the devstack (eg: `/edx/src/`)
1. Set the `TAXONOMY_SAMPLE_PATH` variable inside the `generate.py` file to point to the cloned repo inside your devstack/tutor environment, eg: `/edx/src/taxonomy-sample-data`
1. To run the script, enter the LMS shell (`make lms-shell`) and run the following command:
    ```sh
    python manage.py cms shell < /path/to/taxonomy-sample-data/generate.py
    ```

4. The script should run to completion, you'll notice it tagging the the units (vertical blocks) in the logs of the script

---
Private-ref: [FAL-3523](https://tasks.opencraft.com/browse/FAL-3523)

yusuf-musleh committed:
feat: Add tags to Units

bradenmacdonald reviewed (commented):
One problem with the current script is it just picks the first tag from each taxonomy, which is never a "grandchild" tag, and is always a root tag. It would be much better if most of the tags are leaf tags like `Landscape Architecture` in the LightCast skills taxonomy or `Physical Sciences` in the Canada taxonomy, or `hierarchical taxonomy tag 1.9.13`.

yusuf-musleh committed:
feat: Tag items using leaf tags + refactor

- Tag courses/units/components using leaf tags rather than the first level
tags for better real life representation of data.
- Update tags deletion implementation after implementation changed in
  edx-platform.

yusuf-musleh commented:
@bradenmacdonald Agreed, I think using grandchild (or "leaf") tags are more representative of real life data. I've updated the implementation to tag using the first leaf tag for each taxonomy. I've also updated the tags deletion implementation to account for the changes made in edx-platform related to tags.

bradenmacdonald reviewed (approved):
Nice improvement. Works well!

![Screenshot 2023-11-02 at 1 56 11 PM](https://github.com/open-craft/taxonomy-sample-data/assets/945577/2bb7ef40-628b-4bc9-be55-9e75e999e1ac)
